rest_framework/fields.py

```diff
@@ -903,7 +909,12 @@ def enforce_timezone(self, value):
         When `self.default_timezone` is not `None`, always return aware datetimes.
         """
         if (self.default_timezone is not None) and not timezone.is_aware(value):
-            return timezone.make_aware(value, self.default_timezone)
+            # If a timezone is active, we want to use it, but if not we want to use the timezone
+            # specified for this field over the system's default timezone.
+            if hasattr(timezone._active, 'value'):
+                return from_current_timezone(value)
```

Comment:
You _should_ be able to do

```python
timezone.make_aware(value, timezone._active.value)
```

Which is equivalent to [how `from_current_timezone` works](https://github.com/django/django/blob/7b6d3104f263d9483982928604b2e51f06126ec1/django/forms/utils.py#L162).